packages/signers/multisig-signer/src/multisig-signer.ts

```diff
@@ -0,0 +1,74 @@
+import { getDynamicBuilder, getLookupFn } from "@polkadot-api/metadata-builders"
+import type { PolkadotSigner } from "@polkadot-api/polkadot-signer"
+import {
+  decAnyMetadata,
+  getSs58AddressInfo,
+  SS58String,
+} from "@polkadot-api/substrate-bindings"
+import { toHex } from "@polkadot-api/utils"
+
+export function multisigSigner(
+  multisig: {
+    threshold: number
+    signatories: SS58String[]
+  },
+  signer: PolkadotSigner,
+): PolkadotSigner {
+  const publicAddr = toHex(signer.publicKey)
+  const otherSignatories = multisig.signatories.filter((addr) => {
+    const info = getSs58AddressInfo(addr)
+    if (!info.isValid) throw new Error("Invalid address " + addr)
+    return toHex(info.publicKey) !== publicAddr
+  })
+  if (otherSignatories.length === multisig.signatories.length) {
+    throw new Error("Signer is not one of the signatories of the multisig")
+  }
+
+  return {
+    // TODO should this be the publicKey of the multisig, or the public key of the actual signer?
+    publicKey: signer.publicKey,
```

Comment:
Need to figure this out before merging

Reply:
I think it makes much more sense to have the whole multisig pubkey. Otherwise, getting the multisig address takes more work / other libraries are required.

We should implement the algorithm probably in substrate-bindings.

Reply:
Yeah, I understood the point. I guess it is the public key of the signer it represents. Nevertheless, let's think about that.